### PR TITLE
dist: set `TAR_OPTIONS` to remove ownership info

### PR DESCRIPTION
**Pull request by rauteric**
This resets the file ownership to root in the dist tarballs

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rajachan — DISMISSED** [review]

File: Makefile.am [hunk 1]
```diff
@@ -17,3 +17,6 @@ EXTRA_DIST = \
 	RELEASENOTES.md
 
 AM_DISTCHECK_CONFIGURE_FLAGS=$(NCCL_NET_OFI_DISTCHCK_CONFIGURE_FLAGS)
+
+TAR_OPTIONS = --owner=0 --group=0
```
Comment (rajachan) [inline Makefile.am:21]: This is what the automake manual recommends too, fwiw.
https://www.gnu.org/software/automake/manual/html_node/Basics-of-Distribution.html

Comment (rauteric) [reply Makefile.am:23]: Added that to the comment

**Comment by rauteric** [comment]
bot:aws:retest

**Comment by a-szegel** [comment]
bot:aws:retest

**Pull request merged**